lib/base/process.cpp

```diff
@@ -109,11 +109,11 @@ static Value ProcessSpawnImpl(struct msghdr *msgh, const Dictionary::Ptr& reques
 	int j = 0;
 
 	for (int i = 0; i < envc; i++) {
-		if (strncmp(environ[i], lcnumeric, strlen(lcnumeric)) == 0) {
+		if (environ[i] == lcnumeric) {
 			continue;
 		}
 
-		if (strncmp(environ[i], notifySocket, strlen(notifySocket)) == 0) {
+		if (environ[i] == notifySocket) {
```

Comment:
This now does `==` on `const char*`, i.e. compare pointer values, not what is intended here.

Also, at second view, this doesn't compare the strings for equality but rather checks if `environ[i]` starts with some prefix.

Reply:
🐘